File: thebombermangame.py
#see if a point is out of bounds
def outOfBounds(xPos, yPos, rows, cols):

	thing = xPos >= 0 and xPos < cols and yPos >= 0 and yPos < rows
	return not thing

#print a given vector
def printVector(arr, rows, cols):

	for i in range(rows):
		for j in range(cols):
			print(arr[i][j], end='')
		print("\n", end='')

#explode the bombs in a given grid
def getExploded(inpt, rows, cols):

	arrAlt = [[ord('O') for _x in range(cols)] for _y in range(rows)]

	for i in range(rows):
		for j in range(cols):
			
			if inpt[i][j] == 'O':

				arrAlt[i][j] = ord('.')

				if not (outOfBounds(j-1, i, rows, cols)):
					arrAlt[i][j-1] = ord('.')

				if not (outOfBounds(j+1, i, rows, cols)):
					arrAlt[i][j+1] = ord('.')

				if not (outOfBounds(j, i+1, rows, cols)):
					arrAlt[i+1][j] = ord('.')
					
				if not (outOfBounds(j, i-1, rows, cols)):
					arrAlt[i-1][j] = ord('.')
				
	arrAlt_char = [[chr(x) for x in y] for y in arrAlt]

	return arrAlt_char

rows, cols, seconds = [int(x) for x in input().strip().split()]

arr = []
for arr_i in range(rows):
	arr.append([x for x in list(input().strip())])

arrFull = [['O' for x in range(cols)] for y in range(rows)]

if seconds == 1:
	printVector(arr, rows, cols)
else:
	case = seconds % 4
	if case == 1:
		printVector(getExploded(getExploded(arr, rows, cols), rows, cols), rows, cols)
	elif case == 3:
		printVector(getExploded(arr, rows, cols), rows, cols)
	else:
		printVector(arrFull, rows, cols)

File: thebombermangame.cpp
int rows, columns;

int main() {
    
    int seconds;
    
    cin >> rows >> columns >> seconds;
    
    vector<vector<char>> arr(rows, vector<char>(columns));
    vector<vector<char>> arrFull(rows, vector<char>(columns, 'O'));
    
    for(int i = 0; i < rows; i++){
        for(int j = 0; j < columns; j++){
            cin >> arr[i][j];
        }
    }
    
    if(seconds == 1){
        printVector(arr);
    } else {
        switch (seconds % 4){
            case 1:
                printVector(getExploded(getExploded(arr)));
                break;
            case 3:
                printVector(getExploded(arr));
                break;
            default:
                printVector(arrFull);
        }
    }
    
    return 0;
}

//see if a point is out of bounds
bool outOfBounds(int xPos, int yPos){
    bool thing = (xPos >= 0 && xPos < columns && yPos >= 0 && yPos < rows);
    return !thing;
}

//print a given vector
void printVector(const vector<vector<char>> arr){
    
    for(int i = 0; i < rows; i++){
        for(int j = 0; j < columns; j++){
            cout << arr[i][j];
        }
        cout << endl;
    }
}

//explode the bombs in a given grid
vector<vector<char>> getExploded(const vector<vector<char>>input){
   
    vector<vector<char>> arrAlt(rows, vector<char>(columns,'O'));

    for(int i = 0; i < rows; i++){
        for(int j = 0; j < columns; j++){
            if(input[i][j] == 'O'){
                arrAlt[i][j] = '.';
                
                if(!outOfBounds(j-1, i)){
                    arrAlt[i][j-1] = '.';
                }
                
                if(!outOfBounds(j+1, i)){
                    arrAlt[i][j+1] = '.';
                }
                
                if(!outOfBounds(j, i+1)){
                    arrAlt[i+1][j] = '.';
                }
                
                if(!outOfBounds(j, i-1)){
                    arrAlt[i-1][j] = '.';
                }
            }
        }
    }
    
    return arrAlt;
}